tests/test_serialize.py

```diff
@@ -204,8 +210,31 @@ def test_serialize_ignore_info_wrap():
     class MyModel(BaseModel):
         x: int
 
-        @serializer('x', mode='wrap')
+        @field_serializer('x', mode='wrap')
         def ser_x(v: Any, handler: SerializerFunctionWrapHandler) -> str:
             return f'{handler(v):,}'
 
     assert MyModel(x=1234).model_dump() == {'x': '1,234'}
+
+
+def test_serialize_decorator_self_info():
+    class MyModel(BaseModel):
+        x: Optional[int]
+
+        @field_serializer('x', json_return_type='str')
+        def customise_x_serialisation(self, v, info):
+            return f'{info.mode}:{v:,}'
+
+    assert MyModel(x=1234).model_dump() == {'x': 'python:1,234'}
+    assert MyModel(x=1234).model_dump(mode='foobar') == {'x': 'foobar:1,234'}
+
+
+def test_serialize_decorator_self_no_info():
```

Comment:
This test is failing, @adriangb do you have a quick solution? I looked but couldn't solve it quickly.

Otherwise we can mark it as `xfail` and fix it later.